homeassistant/components/netatmo/sensor.py

```diff
@@ -197,22 +197,23 @@ def _retry(_data):
 class NetatmoSensor(Entity):
     """Implementation of a Netatmo sensor."""
 
-    def __init__(self, netatmo_data, module_name, sensor_type, station):
+    def __init__(self, netatmo_data, module_info, sensor_type):
         """Initialize the sensor."""
-        self._name = "Netatmo {} {}".format(module_name, SENSOR_TYPES[sensor_type][0])
         self.netatmo_data = netatmo_data
-        self.module_name = module_name
+        module = self.netatmo_data.station_data.moduleById(mid=module_info["id"])
+        if module["type"] == "NHC":
+            self.module_name = module_info["station_name"]
+        else:
+            self.module_name = module_info["module_name"]
+        self._name = f"{DOMAIN} {self.module_name} {SENSOR_TYPES[sensor_type][0]}"
```

Comment:
The name changes, right?

Reply:
Yes, should I mark it as a breaking change?

Reply:
If the unique_id doesn't change, the name, and entity_id, can change without causing a breaking change.

Reply:
Actually the names might change for home coach devices. Others should not be affected. I really need to put some tests together to detect such things.